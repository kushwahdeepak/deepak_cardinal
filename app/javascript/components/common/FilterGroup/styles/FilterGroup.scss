@import '../../../../styles/var';

.filters-card {
    color: #474953;
    padding: 25px 35px 27px;
    background: #FFFFFF;
    box-shadow: 0px 4px 15px rgba(0, 0, 0, 0.1);
    border-radius: 15px;
    width: 275px;
    .filters-title {
        font-family: $default-font-family;
        font-style: normal;
        font-weight: 500;
        font-size: 14px;
        line-height: 39px;
        margin-bottom: 3px !important;
        padding: 10px;
        padding-top: 0px !important
    }
    .filterBtn{
        margin-top: 15px;
        font-family: Avenir, "Lato", sans-serif !important;
        font-style: normal !important;
        font-weight: 800 !important;
        font-size: 16px !important;
        line-height: 22px !important;
        color: #ffffff;
        background: linear-gradient(94.67deg, #4C68FF -1.19%, #6077F4 53.94%, #8185FF 102.59%);
        border-radius: 20px !important;
        padding: 9px 27px !important;
    }
}

.candidate-card{
    background: #FFFFFF;
    box-shadow: 0px 4px 15px rgba(0, 0, 0, 0.1);
    border-radius: 15px;
    padding: 2rem;
    margin-top: 44px;
    .filters-title {
        font-family: $default-font-family;
        font-style: normal;
        font-weight: 500;
        font-size: 14px;
        line-height: 39px;
        padding-left: 10px;
        margin-bottom: 3px !important;
        padding: 10px;
        padding-top: 0px !important
    }
    .accordion{
        .collapse{
            margin-top: 0px !important;
            margin-bottom: 0px !important;
        }
    }
}

.accordion{
    padding: 10px;
    .collapse{
        margin-top: 10px !important;
        margin-bottom: 10px !important;
        .form-control{
            background: #FFFFFF !important;
            box-shadow: 0px 4px 5px rgba(0, 0, 0, 0.05) !important;
            border-radius: 5px !important;
        }
        .btn svg{
            margin-right: 0px !important;
        }
        .btn{
            background: #E6E9FC !important;
            border-radius: 25px;
            color: #98A3D9;
            padding: 5px 8px 5px 8px;
        }
        .tagTextInput{
            width: 100%;
            height: calc(1.5em + 0.75rem + 2px);
            padding: 0.375rem 0.75rem;
            font-size: 1rem;
            font-weight: 400;
            line-height: 1.5;
            color: #495057;
            background: #FFFFFF !important;
            box-shadow: 0px 4px 5px rgba(0, 0, 0, 0.05) !important;
            border-radius: 5px !important;
        }
    }
}

.btn-expander{
    color: #474953 !important;
    font-size: 12px !important;
    background-color:none !important;
}
.btn:focus, .btn-primary:focus, .btn-primary:not(:disabled):not(.disabled):active:focus {
    background-color: #4c68ff;
}
.pill{
    padding: 8px !important;
    border: 1px solid #98A3D9;
    border-radius: 20px;
    margin-right: 5px;
    margin-bottom: 5px;
    .pill-close-btn{
        background: none;
    }
    .pill-text{
        padding-left: 10px;
    }
}

.card .card-body{
    background: none !important;
}


@media only screen and (min-device-width : 320px) and (max-device-width : 890px){
    .filters-card {
        width: 100%;
        text-align:center;
        padding: 25px 25px 27px !important;
    }
    .filterBtn{
        margin-left:0px !important;
    }
}

.accordion {
    width:100% !important;
}