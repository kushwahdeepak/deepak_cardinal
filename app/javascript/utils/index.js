import { act } from 'react-dom/test-utils'

export const findByTestAttr = (wrapper, attr) => {
    return wrapper.find(`[data-test='${attr}']`)
}

export const timeoutResolve = (time = 0, response = {}) =>
    new Promise((resolve) => {
        setTimeout(() => resolve(response), time)
    })

export const wait = (time = 0) => {
    const timer = timeoutResolve(time)

    return act(async () => {
        await timer
    })
}
export function emailIsValid(email) {
    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)
}

export function generateRandom(min, max) {
    return Math.floor(min + Math.random() * (max + 1 - min))
}

export function capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase()
}

export function createMarkedList(array) {
    if (!Array.isArray(array)) {
        return []
    }
    return array.map((candidate) => ({
        id: candidate.id,
        name: candidate.id,
        fullName: `${candidate.first_name} ${candidate.last_name}`,
        emailAddress: candidate.email_address,
        isChecked: false,
    }))
}

export function stringReplace(string,toSymbol,fromSymbol){
    return string.replace(toSymbol, fromSymbol)
}

export function capitalize(string) {
    return string.replace(/\b(\w)/g, s => s.toUpperCase());
}

export function skillIsNotEmpty(value){
    return value.trim().length > 0;
}

export function firstCharacterCapital(value){
    return value?.charAt(0)?.toUpperCase() + value?.slice(1)
}
