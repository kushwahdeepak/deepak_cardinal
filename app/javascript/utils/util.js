import isNil from 'lodash.isnil'
import isEmpty from 'lodash.isempty'

class Util {
    /**
     * Define stage values
     */
    static STAGES = ['first_interview', 'second_interview', 'offer', 'reject']

    /**
     * Tests whether a single object attribute includes the at
     * least one string from the given set.
     */
    static attributePassesStringSetFilter(object, attr, stringSet) {
        if (isNil(object[attr])) return false

        // We "OR" the tag tests: the attribute needs to contain only one tag
        return stringSet
            .map((string) => string.trim().toLowerCase())
            .reduce(
                (passes, string) =>
                    passes ||
                    Util.attributeToString(object[attr])
                        .toLowerCase()
                        .includes(string),
                false
            )
    }

    /**
     * Tests whether at least one of a set of object attributes
     * includes every word from a given piece of text.
     */
    static objectPassesTextFilter(object, attrs, text) {
        text = text.trim().toLowerCase()

        if (text === '') return true

        const words = text.split(' ')

        // Join all the given attribute names to lowercase, stringified attribute values
        const allAttrsString = attrs
            .filter((attr) => !isNil(object[attr]))
            .map((attr) => Util.attributeToString(object[attr]).toLowerCase())
            .join('')

        // Is every word included in the allAttrsString?
        return words.reduce(
            (matchesAll, word) => matchesAll && allAttrsString.includes(word),
            true
        )
    }

    static attributeToString(attr) {
        if (typeof attr === 'string') {
            return attr
        } else if (Array.isArray(attr)) {
            return attr.join(',')
        } else if (typeof attr === 'object') {
            return JSON.stringify(attr)
        } else {
            return '' + attr
        }
    }

    static numberWithCommas(x) {
        return x?.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
    }

    /**
     * Returns number of results in format [x] - [y] of [z]
     * @param {number} totalJobsCount Total number of jobs
     * @param {number} pageCount Total number of pages
     * @param {number} activePage Currently active page
     * @param {number} pageSize How many results to display per page
     */
    static displayNumberOfResults(
        totalJobsCount,
        pageCount,
        activePage,
        pageSize,
        totalPersonCount=null,
        candidatePage=false,
        showdisplaycount=true
    ) {
        if (totalJobsCount == 0) return '0 - 0 of 0'
        const offset = pageSize * pageCount - totalJobsCount
        const lastJobIndex =
            (activePage + 1) * pageSize > totalJobsCount
                ? totalJobsCount
                : (activePage + 1) * pageSize
        const firstJobIndex =
            activePage == 0
                ? 1
                : activePage == pageCount - 1
                ? lastJobIndex - pageSize + offset + 1
                : lastJobIndex - pageSize + 1
        return `${firstJobIndex} - ${lastJobIndex} of ${Util.numberWithCommas(
            (candidatePage && showdisplaycount) ? totalPersonCount : totalJobsCount

        )}`
    }

    /**
     * Returns a string indicating the candidate's title and current
     * company, based on available information on the Person object
     * @param {object} candidate Person object from our db
     */
    static candidateCompanyString(candidate) {
        const position = candidate.company_position
        const currentCompany = candidate.current_company

        if (isEmpty(position) && isEmpty(currentCompany)) return ''

        if (!isEmpty(position) && !isEmpty(currentCompany))
            return `${position} at ${currentCompany}`

        if (!isEmpty(position)) return position

        if (!isEmpty(currentCompany)) return `at ${currentCompany}`
    }

    /**
     * Converts the given array of objects with ids to an object whose
     * keys are the ids of the array objects.
     *
     * Every object in the given array must have a unique 'id' property defined.
     *
     * @param {Array} array
     * @returns {Object}
     */
    static arrayToObject(array) {
        const obj = {}
        array.forEach((element) => (obj[element.id] = element))
        return obj
    }

    /**
     * Makes a copy of the given array and replace the value at 'index' with 'value'
     *
     * @param {Array} array
     * @param {Number} index
     * @param {*} value
     * @returns {Array}
     */
    static replaceInArrayCopy_i(array, index, value) {
        const clone = array.slice()
        clone[index] = value
        return clone
    }

    /**
     * Makes a copy of the given array and replaces the element with the same
     * .id property as obj.id with obj
     *
     * @param {Array} array
     * @param {Object} obj
     * @returns {Array}
     */
    static replaceInArrayCopy(array, obj) {
        const index = array.findIndex((element) => element.id === obj.id)
        return Util.replaceInArrayCopy_i(array, index, obj)
    }

    static deleteInArrayCopy(array, index) {
        const copy = array.slice()
        copy.splice(index, 1)

        return copy
    }

    static handleUndefined(value){
        if (value === undefined || value === 'undefined')
        return '';
        else
        return value
    }

    static handleUndefinedFullName(firstName, lastName){
      if(firstName === "undefined" || firstName === undefined)
      return lastName
      else if(lastName === "undefined" || lastName === undefined)
      return firstName
      else
      return firstName + ' ' + lastName     }

    static capitalize(value){
         return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
    }
}

export default Util
