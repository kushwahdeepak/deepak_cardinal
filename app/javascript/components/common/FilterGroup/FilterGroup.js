import React from 'react'
import Card from 'react-bootstrap/Card'
import Button from 'react-bootstrap/Button'
import styles from './styles/FilterGroup.module.scss'
import './styles/FilterGroup.scss'


function FilterGroup({ children, filterCount, experienceYearsCount, candidatePage }) {

    const filtersActive = filterCount > 0

    const experienceYears = experienceYearsCount ? true : ''

    return (

        <Card className={`${styles.filtersPanel + ' shadow-sm'} ${candidatePage ? 'candidate-card' : 'filters-card'}`}>
            <h2 className="filters-title">Filter Your Search:</h2>
            <Card.Body className="p-0">{children}</Card.Body>
        </Card>
    )
}

export default FilterGroup
