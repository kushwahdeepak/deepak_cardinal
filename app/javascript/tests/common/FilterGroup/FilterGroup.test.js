import React from 'react';
import FilterGroup from '../../../components/common/FilterGroup/FilterGroup';
import renderer from 'react-test-renderer';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { findByTestAttr } from '../../../utils';

configure({adapter: new Adapter()});

const initialShallowRender = (props = {}) => {
	return shallow(<FilterGroup {...props}/>);
};

const initialMountRender = (props = {}) => {
	return mount(<FilterGroup {...props}/>);
};

const initialProps = {
	filterCount: 2
};

describe('FilterGroup component',  () => {
	let wrapper;
	let fullWrapper;

	beforeEach(() => {

		const props = {
			...initialProps
		};

		wrapper = initialShallowRender(props);
		fullWrapper = initialMountRender(props);
	});

	afterEach(() => {
		fullWrapper && fullWrapper.unmount();
	});

	it('should render FilterGroup component', () => {
		expect(wrapper.exists()).toBe(true);
	});

	it('should match with a snapshot', () => {
		const component = renderer.create(<FilterGroup />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	describe('FilterGroup component with props', () => {
		let wrapper;
		let fullWrapper;

		beforeEach(() => {

			const props = {
				...initialProps
			};

			wrapper = initialShallowRender(props);
			fullWrapper = initialMountRender(props);
		});

		it('should render Active Span', () => {
			const span = findByTestAttr(fullWrapper, 'filters-active-span');
			expect(span.length).toBe(1);
		});

		it('should render the correct span text', () => {
			const span = findByTestAttr(fullWrapper, 'filters-active-span');
			expect(span.text()).toBe(`${initialProps.filterCount} filters active`);
		});

		it('should have the correct styles for Card Header', () => {
			const CardHeader = findByTestAttr(fullWrapper, 'filters-card-header').at(0);
			expect(CardHeader.prop('style')).toHaveProperty('color', '#6c6ce0');
		});

		it(`should have the correct length of children on CardHeader`, () => {
			const cardHeader = findByTestAttr(fullWrapper, 'filters-card-header').at(1);
			expect(cardHeader.children().length).toBe(2);
		});
	});
	describe('FilterGroup component without props', () => {
		let wrapper;
		let fullWrapper;

		beforeEach(() => {
			wrapper = initialShallowRender();
			fullWrapper = initialMountRender();
		});

		it(`shouldn't render Active Span`, () => {
			const span = findByTestAttr(fullWrapper, 'filters-active-span');
			expect(span.length).toBe(0);
		});
		it(`should have the correct length of children on CardHeader`, () => {
			const cardHeader = findByTestAttr(fullWrapper, 'filters-card-header').at(1);
			expect(cardHeader.children().length).toBe(1);
		});
	});

});
