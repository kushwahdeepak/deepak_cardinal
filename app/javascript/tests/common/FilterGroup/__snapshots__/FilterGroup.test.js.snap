// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterGroup component should match with a snapshot 1`] = `
<div
  className="undefined shadow-sm card"
>
  <div
    className="p-2 d-flex justify-content-between card-header"
    data-test="filters-card-header"
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.35)",
      }
    }
  >
    <button
      className="btn btn-link"
      disabled={false}
      type="button"
    >
      <i
        className="text-danger"
        data-feather="trash-2"
      />
    </button>
  </div>
  <div
    className="p-2 card-body"
  />
</div>
`;
